src/drt/src/dr/FlexDR_init.cpp

```diff
@@ -322,6 +343,295 @@ void FlexDRWorker::initNets_segmentTerms(
+            if (dist < minDist) {
+              minDist = dist;
+              minIndex = j;
+            }
+          }
+        }
+      }
+      return minIndex;
+    }
+    case drcVia: {
+      auto via = static_cast<drVia*>(obj);
+      auto rect = via->getBBox();
+      frSquaredDistance minDist = std::numeric_limits<frSquaredDistance>::max();
+      int minIndex = -1;
+      for (int j = 0; j < connectedComponents.size(); j++) {
+        // component index is j
+        for (auto idx : connectedComponents[j]) {
+          frSquaredDistance dist
+              = getSqrdDist(netOrigGuides[idx].getBBox(), rect);
+          if (dist < minDist) {
+            if (dist == 0) {
+              return j;
+            }
+            minDist = dist;
+            minIndex = j;
+          }
+        }
+      }
+      return minIndex;
+    }
+    case drcPatchWire: {
+      auto wire = static_cast<drPatchWire*>(obj);
+      auto lNum = wire->getLayerNum();
+      auto rect = wire->getBBox();
+      frSquaredDistance minDist = std::numeric_limits<frSquaredDistance>::max();
+      int minIndex = -1;
+      for (int j = 0; j < connectedComponents.size(); j++) {
+        // component index is j
+        for (auto idx : connectedComponents[j]) {
+          if (netOrigGuides[idx].getLayerNum() == lNum
+              && netOrigGuides[idx].intersects(rect)) {
+            return j;
+          } else {
+            frSquaredDistance dist
+                = getSqrdDist(netOrigGuides[idx].getBBox(), rect);
+            if (dist < minDist) {
+              minDist = dist;
+              minIndex = j;
+            }
+          }
+        }
+      }
+      return minIndex;
+    }
+    default:
+      return -1;
+  }
+}
+
+void FlexDRWorker::initNets_initDR_helper(
+    frNet* net,
+    std::vector<std::unique_ptr<drConnFig>>& netRouteObjs,
+    std::vector<std::unique_ptr<drConnFig>>& netExtObjs,
+    std::vector<frBlockObject*> netTerms,
+    std::vector<frRect>& netOrigGuides,
+    std::vector<frRect>& netGuides)
+{
+  struct Node
+  {
+    enum Type
+    {
+      TERM,
+      GUIDE
+    };
+    Type type;
+    int idx;
+    Rect rect;
+    Node(Type typeIn, int idxIn, Rect rectIn)
+        : type(typeIn), idx(idxIn), rect(rectIn)
+    {
+    }
+  };
+  std::vector<Node> nodes;
+  for (int i = 0; i < netGuides.size(); i++) {
+    nodes.push_back(Node(Node::GUIDE, i, netGuides[i].getBBox()));
+  }
+  for (int i = 0; i < netTerms.size(); i++) {
+    Rect rect;
+    if (netTerms[i]->typeId() == frcInstTerm) {
+      auto iterm = static_cast<frInstTerm*>(netTerms[i]);
+      rect = iterm->getBBox(true);
+    } else {
+      auto bterm = static_cast<frBTerm*>(netTerms[i]);
+      rect = bterm->getBBox();
+    }
+    nodes.push_back(Node(Node::TERM, i, rect));
+  }
+  std::map<int, std::vector<int>> nodeMap;
+  for (int i = 0; i < nodes.size(); i++) {
+    for (int j = i + 1; j < nodes.size(); j++) {
+      if (nodes[i].rect.intersects(nodes[j].rect)) {
```

Comment:
Because the DRT modified guides could be disconnected across layers. Meaning that you could have an M2 guide connected to an M4 guide without an M3 guide in between.

Reply:
That sounds like a bug in the guides.  Does grt do this or is it a result of guide process in drt?

Reply:
This is a result of guide process in drt. I haven't looked into the reason yet but I guess it is to make TA assign tracks on M2 and M4 only because M3 is just for via.

Reply:
Wouldn't this generate an out of guide penalty or does that not apply to vias?